app/src/main/java/org/mifos/selfserviceapp/ui/fragments/ClientListFragment.java

```diff
@@ -0,0 +1,145 @@
+        ((BaseActivity) getActivity()).getActivityComponent().inject(this);
+    }
+
+    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle
+            savedInstanceState) {
+        rootView = inflater.inflate(R.layout.fragment_client_list, container, false);
+        setHasOptionsMenu(true);
+        getActivity().setTitle(getResources().getString(R.string.clients_list));
+        ButterKnife.bind(this, rootView);
+
+        mClientListPresenter.attachView(this);
+
+        layoutManager = new LinearLayoutManager(getActivity());
+        layoutManager.setOrientation(LinearLayoutManager.VERTICAL);
+        rv_clients.setLayoutManager(layoutManager);
+        rv_clients.addOnItemTouchListener(new RecyclerItemClickListener(getActivity(), this));
+        rv_clients.setHasFixedSize(true);
+
+        swipeRefreshLayout.setColorSchemeResources(R.color.blue_light, R.color.green_light, R
+                .color.orange_light, R.color.red_light);
+        swipeRefreshLayout.setOnRefreshListener(new SwipeRefreshLayout.OnRefreshListener() {
+            @Override
+            public void onRefresh() {
+                mClientListPresenter.loadClients();
+            }
+        });
+
+        mClientListPresenter.loadClients();
+
+        return rootView;
+    }
+
+    @Override
+    public void onDestroyView() {
+        mClientListPresenter.detachView();
+        super.onDestroyView();
+    }
+
+    @Override
+    public void onItemClick(View childView, int position) {
+        //TODO add client accounts screen from here
+    }
+
+    @Override
+    public void onItemLongPress(View childView, int position) {
+
+    }
+
+    @Override
+    public void showProgress() {
+        if (progress == null) {
+            progress = new ProgressDialog(getActivity(), ProgressDialog.STYLE_SPINNER);
+            progress.setCancelable(false);
+        }
+        progress.setMessage(getResources().getText(R.string.progress_message_loading));
+        progress.show();
+    }
+
+    @Override
+    public void hideProgress() {
+        if (progress != null && progress.isShowing())
+            progress.dismiss();
+    }
+
+    @Override
+    public void showClients(List<Client> clientList) {
+        this.clientList = clientList;
+        inflateClientList();
+        if (swipeRefreshLayout.isRefreshing())
+            swipeRefreshLayout.setRefreshing(false);
+    }
+
+    /**
+     * setting adapter to display list of clients on the client list screen
+     */
+    private void inflateClientList() {
+        clientListAdapter = new ClientListAdapter(getContext(), clientList);
+        rv_clients.setAdapter(clientListAdapter);
+    }
+
+    @Override
+    public void showErrorFetchingClients(String message) {
+        //TODO: Handle this error properly
+        Toast.makeText(getActivity(), message, Toast.LENGTH_LONG).show();
+    }
+
+    @Override
+    public void showInternalServerError(String message) {
```

Comment:
Oh. Yes didn't think that way. Thanks, will do that.

On Jul 21, 2016 6:22 PM, "Ishan Khanna" <notifications@github.com> wrote:

> In
> app/src/main/java/org/mifos/selfserviceapp/ui/fragments/ClientListFragment.java
> <https://github.com/openMF/self-service-app/pull/3#discussion_r71698903>:
>
> > +    /**
> > +     * setting adapter to display list of clients on the client list screen
> > +     */
> > +    private void inflateClientList() {
> > +        clientListAdapter = new ClientListAdapter(getContext(), clientList);
> > +        rv_clients.setAdapter(clientListAdapter);
> > +    }
> > +
> > +    @Override
> > +    public void showErrorFetchingClients(String message) {
> > +        //TODO: Handle this error properly
> > +        Toast.makeText(getActivity(), message, Toast.LENGTH_LONG).show();
> > +    }
> > +
> > +    @Override
> > +    public void showInternalServerError(String message) {
>
> @vjs3 <https://github.com/vjs3> That is fine. I get that you are showing
> different error messages which is very nice and makes the user experience
> much better.
>
> But I don't understand the need of two different methods for this.
>
> Can't we just have one method on the view called showError(String message)
>
> and then from within the presenter you can call it whenever you have an
> error
> like getMvpView().showError("Failed to load client list") or getMvpView().showError("Could
> not authenticate user") etc.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/openMF/self-service-app/pull/3/files/6870d4f8418c11cb3c006984abf805d1d8459f58#r71698903>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AK4JNoMv0rdZqK7jHOTwd7kp_mpBQDHHks5qX2uNgaJpZM4JOAOM>
> .
>